util/dvsim/Deploy.py

```diff
@@ -216,7 +243,12 @@ def is_equivalent_job(self, item):
         return True
 
     def dispatch_cmd(self):
-        self.exports.update(os.environ)
+        # Merge the shell's env variables with exports as long as they are
+        # unique
+        self.exports.update(
+            {k: v
+             for k, v in os.environ.items() if k not in self.exports.keys()})
```

Comment:
Good catch! I'm amazed that this hadn't broken something already :-)

It's probably clearer to do this the other way around:
```python
merged = os.environ.copy()
merged.update(self.exports)
self.exports = merged
```
It seems that Python 3.9 finally includes a "pure" version so you can write
```python
self.exports = os.environ() | self.exports
```
but I guess it will be a while before we can use that!

However, thinking about it, it seems a little odd that we're updating `self.exports` (the list of things we read from the hjson) at all. Can't we do this instead?
```python
exports = os.environ.copy()
exports.update(self.exports)
```
and then use `exports` instead of `self.exports` when dumping environment variables and spawning the subprocess?

Reply:
Sure, updated.